src/main/java/net/mcreator/element/parts/gui/Checkbox.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2020 Pylo and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.element.parts.gui;
+
+import net.mcreator.element.parts.Procedure;
+import net.mcreator.minecraft.MinecraftImageGenerator;
+import net.mcreator.ui.wysiwyg.WYSIWYGEditor;
+import net.mcreator.workspace.Workspace;
+
+import java.awt.*;
+
+public class Checkbox extends GUIComponent implements IMachineNamedComponent {
+	public String text;
+	public Procedure isCheckedProcedure;
+
+	public Checkbox(String name, int x, int y, String text, Procedure isCheckedProcedure) {
+		super(name, x, y);
+		this.text = text;
+		this.isCheckedProcedure = isCheckedProcedure;
+	}
+
+	@Override public void paintComponent(int cx, int cy, WYSIWYGEditor wysiwygEditor, Graphics2D g) {

```

Comment:
This preview needs to be improved so size/border width of checkbox matches MC + there is no text included in the preview which is not ok.

@Goldorion and @MysteriousWolf please arrange who will improve this.

Reply:
I only made a basic thing o have something, but I didn’t want to have it in the final version.

Reply:
paintComponent to show string can be done by you I think, also preview here, it is two rectangles :D

Reply:
Oh ok, I thought it was a preview image or the preview inside the WYSIWYG editor.

Reply:
Preview inside WYSIWYG is paintComponent, Matej will do preview image on the button